src/env.rs

```diff
@@ -101,7 +100,8 @@ pub fn get_partition_ctx(
     yb8: c_int,
     xb8: c_int,
 ) -> u8 {
-    (a.partition[xb8 as usize] >> (4 - bl) & 1) + ((l.partition[yb8 as usize] >> (4 - bl) & 1) << 1)
+    (a.partition[xb8 as usize] >> (bl.invert() as u8) & 1)
+        + ((l.partition[yb8 as usize] >> (bl.invert() as u8) & 1) << 1)
```

Comment:
this logic is kind of nasty: what does this mean? I'm guessing the partition just stores the value in the logical order 8 .. 128 while now the order becomes 128 .. 8?

Reply:
actually, now that I look at this again, the associativity is different, the expression is really 

```
((a.partition[xb8 as usize] >> (bl.invert() as u8)) & 1)
```

the shift happens before the bitwise and, which is consistent with https://doc.rust-lang.org/reference/expressions.html#expression-precedence, `>>` has higher precedence than `&`.